rasa/shared/core/events.py

```diff
@@ -322,7 +342,23 @@ def __init__(
         input_channel: Optional[Text] = None,
         message_id: Optional[Text] = None,
         metadata: Optional[Dict] = None,
+        use_text_for_featurization: Optional[bool] = None,
```

Comment:
does this really need to be stored here? why does this not get computed during the featurization if it is needed there?

Reply:
because it is computed on previous prediction state per conversation, and the tracker is the only one that contains such info, and the tracker dialogue history is represented as events

Reply:
can't this be stored alongside the tracker during the featurization? it seems odd to store this on the utterance, esp. if you think about other featurization algorithms that might think differently about the use of the text

Reply:
but then we'd create separate list that should be aligned with the list of events? we discussed that possibility but removed that since it will be hard to maintain two separate lists that should be strongly aligned, also this info cannot be recreated. it should be persisted together with the tracker

Reply:
I think that we need to refactor how the no-ML parts of the code interface with the ML-parts of the code. One solution could be to have strong interfaces between these parts and I also think that having a "ML version" of certain objects (e.g. `Tracker` but also things like `Domain` which is currently half ML featurizer / half Domain object) can be a part of this. 

I think changing this should be part of a larger refactoring though as the `DialogueStateTracker` is just one part where this is happening.

Reply:
can you make sure we keep track of this in the issues?

Reply:
Will do. Big issue creation day tomorrow 👍🏻